src/Ty.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+namespace Dhii\Structs;
+
+use ArrayAccess;
+use Dhii\Structs\PropTypes\ArrayPropType;
+use Dhii\Structs\PropTypes\BoolPropType;
+use Dhii\Structs\PropTypes\CallablePropType;
+use Dhii\Structs\PropTypes\CustomPropType;
+use Dhii\Structs\PropTypes\EnumPropType;
+use Dhii\Structs\PropTypes\FloatPropType;
+use Dhii\Structs\PropTypes\IntersectionPropType;
+use Dhii\Structs\PropTypes\IntPropType;
+use Dhii\Structs\PropTypes\IterablePropType;
+use Dhii\Structs\PropTypes\NullablePropType;
+use Dhii\Structs\PropTypes\ObjectPropType;
+use Dhii\Structs\PropTypes\StringPropType;
+use Dhii\Structs\PropTypes\UnionPropType;
+use TypeError;
+
+/**
+ * Helper methods for easily declaring property types.
+ *
+ * Some of the below methods use a static cache to conserve memory. Types that require or can accept arguments will not
+ * have their instances cached.
+ *
+ * Also provides additional utility methods for property type implementations.
+ *
+ * @since [*next-version*]
+ */
+class Ty
```

Comment:
I understand, this class has no interface because it's all static, which means you would couple anyhow. But maybe we can have instead a factory with identical instance methods?

Reply:
That would require struct declarations to require a factory instance, which is very inconvenient. Plus I'm very much against factories in general.

Reply:
Also, in order to be able to do `new MyStruct([...])`, the struct class must be aware of property types statically. That makes it impossible for a factory instance to be used.